openstack/dns/v2/recordsets/requests.go

```diff
@@ -60,3 +60,104 @@ func Get(client *gophercloud.ServiceClient, zoneID string, rrsetID string) (r Ge
 	_, r.Err = client.Get(rrsetURL(client, zoneID, rrsetID), &r.Body, nil)
 	return
 }
+
+// CreateOptsBuilder allows extensions to add additional attributes to the Create request.
+type CreateOptsBuilder interface {
+	ToRecordSetCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts specifies the base attributes that may be used to create a RecordSet.
+type CreateOpts struct {
+	// Name is the name of the RecordSet.
+	Name string `json:"name,required"`
+
+	// Description is a description of the RecordSet.
+	Description string `json:"description,omitempty"`
+
+	// Records are the DNS records of the RecordSet.
+	Records []string `json:"records,omitempty"`
+
+	// TTL is the time to live of the RecordSet.
+	TTL int `json:"-"`
+
+	// Type is the RRTYPE of the RecordSet.
+	Type string `json:"type,omitempty"`
+}
+
+// ToRecordSetCreateMap formats an CreateOpts structure into a request body.
+func (opts CreateOpts) ToRecordSetCreateMap() (map[string]interface{}, error) {
+	b, err := gophercloud.BuildRequestBody(opts, "")
+	if err != nil {
+		return nil, err
+	}
+
+	if opts.TTL > 0 {
```

Comment:
I think this block can be removed by adding `ttl,omitempty` to the `TTL` field in `CreateOpts`.